document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -388,6 +399,52 @@ public CloseableIterator<Document> aggregate(
     return executeQueryV1(query);
   }
 
+  @Override
+  public Optional<Document> update(
+      final org.hypertrace.core.documentstore.query.Query query,
+      final java.util.Collection<SubDocumentUpdate> updates)
+      throws IOException {
+
+    try (final Connection connection = client.getNewConnection()) {
+      connection.setAutoCommit(false);
+
+      org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser parser =
+          new org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser(
+              collectionName, query);
+      final String selectQuery = queryBuilder.buildSelectQueryForUpdateSkippingLocked(parser);
+
+      try (final PreparedStatement pStmt = connection.prepareStatement(selectQuery)) {
```

Comment:
Q: Does this final query implicitly put `LIMIT 1` clause if it is not present in client call of the main query request?

Reply:
Yes. Refer. line number 28 in org.hypertrace.core.documentstore.postgres.PostgresQueryBuilder.